From 23d655ada5bb3b69cd25ef2934bf0ef1a6ebd29e Mon Sep 17 00:00:00 2001
From: gbhakteX <gurudattax.bhakte@intel.com>
Date: Thu, 26 Jul 2012 16:10:40 +0530
Subject: [PATCH 47/56] Ending time of RTSP streaming to show ""-.--.--"
 instead of "0:00".

BZ: 35440

Setting -1 for rtsp streaming contents which are unable to provide valid duration
information.

Change-Id: Iad4c6e8d8bdcb0439f1fa9e97dd0a59655197de2
Orig-Change-Id: I6570815496d2980612ca67c65618e3d1b41df7e5
Category: aosp improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: internal
Upstream-Candidate: yes
Signed-off-by: gbhakteX <gurudattax.bhakte@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libmediaplayerservice/nuplayer/RTSPSource.cpp | 2 +-
 media/libstagefright/rtsp/APacketSource.cpp         | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
index 726f2ff..3a32399 100644
--- a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
+++ b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
@@ -293,7 +293,7 @@ void NuPlayer::RTSPSource::setEOSTimeout(bool audio, int64_t timeout) {
 }
 
 status_t NuPlayer::RTSPSource::getDuration(int64_t *durationUs) {
-    *durationUs = 0ll;
+    *durationUs = -1ll;
 
     int64_t audioDurationUs;
     if (mAudioTrack != NULL
diff --git a/media/libstagefright/rtsp/APacketSource.cpp b/media/libstagefright/rtsp/APacketSource.cpp
index 462c384..915a652 100644
--- a/media/libstagefright/rtsp/APacketSource.cpp
+++ b/media/libstagefright/rtsp/APacketSource.cpp
@@ -420,7 +420,7 @@ APacketSource::APacketSource(
     if (sessionDesc->getDurationUs(&durationUs)) {
         mFormat->setInt64(kKeyDuration, durationUs);
     } else {
-        mFormat->setInt64(kKeyDuration, 60 * 60 * 1000000ll);
+        mFormat->setInt64(kKeyDuration, -1);
     }
 
     mInitCheck = OK;
-- 
2.7.4

